code/game/mecha/mecha.dm

```diff
@@ -1945,6 +1945,20 @@
 /obj/mecha/acidable()
 	return 0
 
+/obj/mecha/beam_connect(var/obj/effect/beam/B)
+	..()
+	apply_beam_damage(B)
+
+
+/obj/mecha/beam_disconnect(var/obj/effect/beam/B)
```

Comment:
Er, sure that's correct?

Reply:
It's precisely how normal mobs take damage from it.